src/irmin/conf.ml

```diff
@@ -166,7 +166,7 @@ let equal t1 t2 =
 
 (* ~root *)
 let root spec =
-  key ~allow_duplicate:true ~spec ~docv:"ROOT"
+  key ~spec ~docv:"ROOT"
```

Comment:
Can we also remove the optional `?allow_duplicate` parameter from the function `key` since it's always a bad idea to overwrite an existing key? (I think it was only added because of the root key issue that you have now fixed https://github.com/mirage/irmin/pull/2252 )